### PR TITLE
OKTA-459959 Adding the Embedded Sign-In: SPA with Embedded SIW - Angular 

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Adding the Angular Guide embedded SIW OIE guide with the new navigation structure (disregard PR#2963)
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-459959](https://oktainc.atlassian.net/browse/OKTA-459959)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+The following sample application code renders the Sign-In widget and after a successful authentication receives an object with OAuth tokens. From the `login.components.ts` file:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md:1]:
```suggestion
The following sample application code renders the Sign-In Widget, and after a successful authentication, receives an object with OAuth tokens. From the `login.components.ts` file:
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+```javascript
+ngOnInit() {
+    // When navigating to a protected route, the route path will be saved as the `originalUri`
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md:5]:
```suggestion
    // When navigating to a protected route, the route path is saved as the `originalUri`
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    // If no `originalUri` has been saved, then redirect back to the app root
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md:6]:
```suggestion
    // If no `originalUri` is saved, then redirect back to the app root
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      // In this flow the redirect to Okta occurs in a hidden iframe
+      this.oktaAuth.handleLoginRedirect(tokens);
+    }).catch((err: any) => {
+      // Typically due to misconfiguration
+      throw err;
+    });
+  }
+  ```
+
+The following sample application code in the `home.component.ts` file retrieves the user information from the ID Token:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md:28]:
```suggestion
The following sample application code in the `home.component.ts` file retrieves the user information from the ID token:
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+async ngOnInit() {
+    const isAuthenticated = await this.oktaAuth.isAuthenticated();
+    if (isAuthenticated) {
+      const userClaims = await this.oktaAuth.getUser();
+      this.userName = userClaims.name as string;
+    }
+  }
+  ```
+
+And displays the user's name on the Custom Login Page after you sign in.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md:40]:
```suggestion
And displays the user's name on the Custom Login Page after you sign in
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+### Routes
+
+Some application routes require authentication in order to render. Define these protected routes with the [`OktaAuthGuard`](https://github.com/okta/okta-angular#oktaauthguard) method from the `@okta/okta-angular` SDK. For example, in the sample application, the Profile page is protected using this method:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/basic-sign-in.md:44]:
```suggestion
Some application routes require authentication to render. Define these protected routes with the [`OktaAuthGuard`](https://github.com/okta/okta-angular#oktaauthguard) method from the `@okta/okta-angular` SDK. For example, in the sample application, the Profile page is protected using this method:
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+The sample application requires an integration with your Okta org to implement authentication for your application users. You can create an Okta custom app integration from the Admin Console, and update the sample application with your configuration details in the following sections.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:1]:
```suggestion
The sample application requires an integration with your Okta org to implement authentication for your application users. You can create an Okta custom app integration from the Admin Console and update the sample application with your configuration details in the following sections.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+1. In the Admin Console, go to **Applications** > **Applications**.
+1. Click **Create App Integration**.
+1. Select **OIDC - OpenID Connect** as the **Sign-in method**.
+1. Select **Single-Page Application** for the **Application Type**.
+1. On the **New Single-Page App Integration** page:
+
+   * Enter an application name.
+   * Select the **Interaction Code** check box.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:10]:
```suggestion
   * Select the **Interaction Code** checkbox.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+   * Select the **Refresh Token** check box.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:11]:
```suggestion
   * Select the **Refresh Token** checkbox.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+   * Set **Base URI** to `http://localhost:8080`
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:12]:
```suggestion
   * Set **Base URI** to `http://localhost:8080`.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+   * Set **Sign-in redirect URIs** to `http://localhost:8080/login/callback`.
+   * Set **Sign-out redirect URIs** to `http://localhost:8080/`
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:14]:
```suggestion
   * Set **Sign-out redirect URIs** to `http://localhost:8080/`.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+1. In the **Assignments** section, select **Allow everyone in your organization to access**.
+1. Click **Save**.
+1. Select the **Sign On** tab.
+1. On the **Sign On Policy** tab, verify that the **Available Authenticators** settings are appropriate for your app. For this use case, ensure that the **1 factor type** authenticator is **Password / IdP**.
+1. On the **Security** > **API** > **Authorization Servers** page, verify that the custom authorization server uses the Interaction Code grant type by selecting the **default** server, clicking **Access Policies**, and editing the **Default Policy Rule**. Review the **If Grant type is** section to ensure the **Interaction Code** check box is selected.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:20]:
```suggestion
1. On the **Security** > **API** > **Authorization Servers** page, verify that the custom authorization server uses the Interaction Code grant type by selecting the **default** server, clicking **Access Policies**, and editing the **Default Policy Rule**. Review the **If Grant type is** section to ensure the **Interaction Code** checkbox is selected.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+1. On the **Security** > **API** > **Trusted Origins** page, ensure there is an entry for your sign in redirect URI. See [Enable CORS](/docs/guides/enable-cors/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/create-app-integration.md:21]:
```suggestion
1. On the **Security** > **API** > **Trusted Origins** page, ensure that there is an entry for your sign-in redirect URI. See [Enable CORS](/docs/guides/enable-cors/).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/install-sdk.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
+1. In your project folder, download the sample application from the gitHub repository, and navigate to the custom-login folder.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/install-sdk.md:1]:
```suggestion
1. In your project folder, download the sample application from the gitHub repository and navigate to the custom-login folder.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/install-sdk.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    cd samples-js-angular/custom-login
+    ```
+
+1. In the `custom-login` folder, install the dependencies:
+
+    ```bash
+    npm install
+    ```
+
+1. Ensure you're using the angular version and node version designed for this sample application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/install-sdk.md:14]:
```suggestion
1. Ensure that you're using the angular version and node version designed for this sample application.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/install-sdk.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    ng --version
+    ```
+
+1. Install the following version of `okta-auth-js` SDK:
+
+    ```bash
+    npm install @okta/okta-auth-js@5.9.1
+    ```
+
+1. Navigate to the `samples-js-angular` folder, and create a configuration file, `testenv` (no extension).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/install-sdk.md:30]:
```suggestion
1. Navigate to the `samples-js-angular` folder and create a configuration file called `testenv` (no extension).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/intro.md [hunk 17]
```diff
@@ -0,0 +1,7 @@
+The embedded Okta Sign-In Widget provides user authentication, as well as an opportunity to customize the sign in experience for your Angular application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/intro.md:1]:
```suggestion
The embedded Okta Sign-In Widget provides user authentication, as well as an opportunity to customize the sign-in experience for your Angular application.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/intro.md [hunk 18]
```diff
@@ -0,0 +1,7 @@
...
+
+This guide explains how to build a sample application with the embedded Sign-In Widget included in the [okta/samples-js-angular](https://github.com/okta/samples-js-angular) repository, and reviews code snippets you can use for your Angular application's authentication  needs. The sample application uses the embedded Widget to sign in to a simple web page that provides details on the user signed in.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/intro.md:3]:
```suggestion
This guide explains how to build a sample application with the embedded Sign-In Widget that is included in the [okta/samples-js-angular](https://github.com/okta/samples-js-angular) repository, and reviews code snippets that you can use for your Angular application's authentication needs. The sample application uses the embedded Widget to sign in to a simple web page that provides details on the user signed in.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+
+### To use the CDN, include the following tags in your HTML:
+
+```html
+<!-- Latest CDN production JavaScript and CSS -->
+<script src="https://global.oktacdn.com/okta-signin-widget/-=OKTA_REPLACE_WITH_WIDGET_VERSION=-/js/okta-sign-in.min.js" type="text/javascript"></script>
+<link href="https://global.oktacdn.com/okta-signin-widget/-=OKTA_REPLACE_WITH_WIDGET_VERSION=-/css/okta-sign-in.min.css" type="text/css" rel="stylesheet"/>
+```
+
+More information, including the latest published version, is available in the [Okta Sign-In Widget Documentation](https://github.com/okta/okta-signin-widget#using-the-okta-cdn).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:11]:
```suggestion
More information, including the latest published version, is available in the [Okta Sign-In Widget documentation](https://github.com/okta/okta-signin-widget#using-the-okta-cdn).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+
+### npm
+
+```bash
+# Run this command in your project root folder.
+npm install @okta/okta-signin-widget@-=OKTA_REPLACE_WITH_WIDGET_VERSION=-
+```
+
+More information, including the latest published version, is available in the [Okta Sign-In Widget Documentation](https://github.com/okta/okta-signin-widget#using-the-npm-module).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:20]:
```suggestion
More information, including the latest published version, is available in the [Okta Sign-In Widget documentation](https://github.com/okta/okta-signin-widget#using-the-npm-module).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+  signIn: any;
+  constructor(public oktaAuth: OktaAuth) {
+    this.signIn = new OktaSignIn({
+      /**
+       * Note: when using the Sign-In Widget for an OIDC flow, it still
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:44]:
```suggestion
       * Note: when using the Sign-In Widget for an OIDC flow, you still
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+       * needs to be configured with the base URL for your Okta Org. Here
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:45]:
```suggestion
       * need to configure it with the base URL for your Okta Org. Here
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+          'primaryauth.title': 'Sign in to Angular & Company',
+        },
+      },
+      authClient: oktaAuth,
+      useInteractionCodeFlow: sampleConfig.widget.useInteractionCodeFlow === 'true',
+    });
+  }
+```
+
+And references the Okta app configuration information in `app.config.js` or `env.js` and the `testenv` file:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:63]:
```suggestion
And references the Okta app configuration information in `app.config.js` or `env.js` file and the `testenv` file:
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/next-steps.md [hunk 24]
```diff
@@ -0,0 +1,3 @@
+You have now successfully authenticated with Okta! Now what? With a user's `id_token`, you have basic claims for the user's identity. You can extend the set of claims by modifying the `scopes` to retrieve custom information about the user. This includes `locale`, `address`, `groups`, and [more](/docs/reference/api/oidc/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/next-steps.md:1]:
```suggestion
You have now successfully authenticated with Okta. Now what? With a user's `id_token`, you have basic claims for the user's identity. You can extend the set of claims by modifying the `scopes` to retrieve custom information about the user. This includes `locale`, `address`, `groups`, and [more](/docs/reference/api/oidc/).
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/next-steps.md [hunk 25]
```diff
@@ -0,0 +1,3 @@
...
+
+Want to learn how to use the user's `access_token`? Check out our [how-to guide](/docs/guides/sign-into-spa/angular/main/) to learn about protecting routes on your server, validating the `access_token`, and more!
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/next-steps.md:3]:
```suggestion
Want to learn how to use the user's `access_token`? Check out our [how-to guide](/docs/guides/sign-into-spa/angular/main/) to learn about protecting routes on your server, validating the `access_token`, and more.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/nutrition.md [hunk 26]
```diff
@@ -0,0 +1,20 @@
+This guide explains how to sign in to an Angular framework single page application (SPA) with the embedded Okta Sign-In Widget.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/nutrition.md:1]:
```suggestion
This guide explains how to sign in to an Angular framework single-page application (SPA) with the embedded Okta Sign-In Widget.
```

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/run-sample.md [hunk 27]
```diff
@@ -0,0 +1,13 @@
...
+
+1. From the `custom-login` folder, run the sample application:
+
+    ```bash
+    ng serve
+    ```
+
+1. After a successful compilation, navigate to `http://localhost:8080`. The **Custom Login with Sign-In Widget** page appears.
+
+1. Click the **Login** button, and enter credentials for a user assigned to your app integration. The **Custom Login with Sign-In Widget** returns with the user's email address.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/run-sample.md:11]:
```suggestion
1. Click **Login** and enter credentials for a user assigned to your app integration. The **Custom Login with Sign-In Widget** returns with the user's email address.
```

**Review by susanharper-okta — COMMENTED** [review]
couple things. looking good!

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md (hunk 19 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md]: I changed this to `Okta Sign-In Widget Documentation`, which is the official title of the repo

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md (hunk 23 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:112]: They're all file here, so, it's a bit clunky if I mention that. How about: 

_And references the Okta app configuration information in `app.config.js` or in the 'env.js` and `testenv` files._

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/next-steps.md (hunk 25 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/next-steps.md]: I removed this section as the reference will no longer be accurate.

File: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md (hunk 22 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/angular/load-app.md:94]: recast as follows: "you still need to configure the base URL for your Okta Org."

**Comment by brianduffield-okta** [comment]
Merging into OIE-by-Default branch

**Pull request merged**